src/main/java/com/zaxxer/hikari/metrics/MetricsTrackerFactory.java

```diff
@@ -26,4 +31,13 @@
     * @return a IMetricsTracker implementation instance
     */
    IMetricsTracker create(String poolName, PoolStats poolStats);
+
+   static MetricsTrackerFactory from(MetricRegistry metricRegistry) {
```

Comment:
I don't think MetricsTrackerFactory should contain references to implementation specific code.

Reply:
I didn't want to create MetricsTrackerFactoryFactory :)
This seems not right at first but in JDK itself starting from 9 there are lots of examples like this. For example [List.of(...)](https://github.com/netroby/jdk9-dev/blob/master/jdk/src/java.base/share/classes/java/util/List.java#L856-L860)

Do you want me to revert this file so that users have to do `setMetricsTrackerFactory(new CodahaleMetricsTrackerFactory(metricRegistry))` instead?

Reply:
There is no risk of CNFE here if that's your concern. This method cannot be invoked without creating MetricRegistry first. Which means the class will already be loaded by JVM before reaching this point. 
When deprecating corresponding metricsTracker implementation this factory method can be deprecated at the same time.

Reply:
More concerned about it being too tightly coupled.